src/agent/common.ts

```diff
@@ -437,21 +438,7 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
         }
     });
 
-    if (maskHints.length === 0) {
-        return noReplacement;
-    }
-    else if (maskHints.length === 1) {
-        var maskHint = maskHints[0];
-        if (maskHint.type === agentifm.MaskType.Variable) {
-            var toReplace = jobEnvironment.variables[maskHint.value];
-            return (input: string) => {
-                return input.replace(toReplace, MASK_REPLACEMENT);
-            };
-        }
-        return noReplacement;
-    }
-    else {
-        // multiple strings to replace
+    if (endpoints != null || maskHints.length > 0) {
```

Comment:
also check endpoints.length > 0? given you are initializing endpoints to empty list, it wouldn't be null & you can only check for length?